playwright/_impl/_frame.py

```diff
@@ -689,17 +690,21 @@ async def set_input_files(
         timeout: float = None,
         noWaitAfter: bool = None,
     ) -> None:
-        params = locals_to_params(locals())
         converted = await convert_input_files(files, self.page.context)
-        if converted["files"] is not None:
-            await self._channel.send(
-                "setInputFiles", {**params, "files": converted["files"]}
-            )
-        else:
-            await self._channel.send(
-                "setInputFilePaths",
-                locals_to_params({**params, **converted, "files": None}),
-            )
+        await self._channel.send(
+            "setInputFiles",
+            {
+                **filter_none(
```

Comment:
I might follow up and do it like in .NET that we remove None by default in send().